wgsl/index.bs

```diff
@@ -5231,6 +5231,76 @@ See [[#builtin-inputs-outputs]] for how to declare a built-in variable.
   </xmp>
 </div>
 
+# Operators # {#operators}
+
+[SHORTNAME] has symbol-based operators, such as the `+` in `a + b`.
+
+## Arithmatic operators ## {#arithmatic-operators}
+
+<table class='data'>
+  <caption>Binary operators</caption>
+  <thead>
+    <tr><th>Category<th>Preconditions<th>Conclusions
+  </thead>
+  <tr>
+    <td>Scalar arithmatic
+    <td>S: f32 or i32 or u32
+    <td>
+      <ul>
+        <li>`(S + S) -> S` (e.g. OpFAdd)
+        <li>`(S - S) -> S` (e.g. OpFSub)
+        <li>`(S * S) -> S` (e.g. OpFMul)
+        <li>`(S / S) -> S` (e.g. OpFDiv)
+        <li>`(S % S) -> S` (e.g. OpFMod)
+      </ul>
+  <tr>
+    <td>Vector arithmatic
+    <td>V: vecN<S>; S: f32 or i32 or u32
+    <td>
+      <ul>
+        <li>`(V + V) -> V` (component-wise, e.g. OpFAdd)
+        <li>`(V - V) -> V` (component-wise, e.g. OpFSub)
+        <li>`(V * V) -> V` (component-wise, e.g. OpFMul)
+        <li>`(V / V) -> V` (component-wise, e.g. OpFDiv)
+        <li>`(V % V) -> V` (component-wise, e.g. OpFMod)
+      </ul>
+  <tr>
+    <td>Vector OP Scalar "splats"
+    <td>V: vecN<S>; S: f32 or i32 or u32
+    <td>
+      <ul>
+        <li>`(V + S) -> (V    + V(S)) -> V`
+        <li>`(S + V) -> (V(S) + V   ) -> V`
+        <li>`(V - S) -> (V    - V(S)) -> V`
+        <li>`(S - V) -> (V(S) - V   ) -> V`
+        <li>`(V * S) -> (V    * V(S)) -> V` (e.g. OpVectorTimesScalar)
+        <li>`(S * V) -> (V(S) * V   ) -> V` (e.g. OpVectorTimesScalar)
+        <li>`(V / S) -> (V    / V(S)) -> V`
+        <li>`(S / V) -> (V(S) / V   ) -> V`
+        <li>`(V % S) -> (V    % V(S)) -> V`
+        <li>`(S % V) -> (V(S) % V   ) -> V`
+      </ul>
+  <tr>
+    <td>Matrix arithmatic
+    <td>S: f32
+    <td>
+      <ul>
+        <li>`(matAxB<S> + matAxB<S>) -> matAxB<S>` (component-wise)
+        <li>`(matAxB<S> - matAxB<S>) -> matAxB<S>` (component-wise)
+        <li>`(matAxB<S> * matCxA<S>) -> matCxB<S>` (linear-algebraic, e.g. OpMatrixTimesMatrix)
+        <li>`(matAxB<S> * vecA<S>) -> vecB<S>` (linear-algebraic, vec as column-vector, e.g. OpMatrixTimesVector)
+        <li>`(vecB<S> * matAxB<S>) -> vecA<S>` (linear-algebraic, vec as row-vector, e.g. OpVectorTimesMatrix)
+      </ul>
+  <tr>
+    <td>Matrix OP Scalar "splats"
+    <td>M: matAxB<S>; S: f32
+    <td>
+      <ul>
+        <li>`(M * S) -> (M    * M(S)) -> M` (e.g. OpMatrixTimesScalar)
```

Comment:
for the spec, I don't think `M(S)` or `V(S)` are needed. We should just say it's a linear algebra scale

Reply:
I will clarify this, because it's *not* linear-algebraic, but rather component-wise. (`mat2x2(S, S, S, S)` not `mat2x2(S, 0, 0, S)`)

Reply:
Hmm, maybe it is. It scales component-wise, which is mat2x2(s, 0, 0, s), isn't it! My bad!